services/venice-controller/src/main/java/com/linkedin/venice/controller/VeniceController.java

```diff
@@ -238,24 +237,19 @@ AdminSparkServer createAdminServer(boolean secure) {
         secure ? secureRequestHandler : unsecureRequestHandler);
   }
 
-  private TopicCleanupService createTopicCleanupService() {
+  private Optional<TopicCleanupService> createTopicCleanupService() {
```

Comment:
I think we should not remove TCS yet. It should be done once we fully migrate to version status based tracking.

Reply:
I think we can keep this code untouched

Reply:
What's the point of making this optional? Can we just add a comment and revert back to the original state